File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/2996-smallest-missing-integer-greater-than-sequential-prefix-sum.cpp
class Solution {
public:
    int missingInteger(vector<int>& nums) {
        set<int> st;
        int n=nums.size();
        int sum=nums[0];
        for(int i:nums){
            st.insert(i);
        }
        for(int i=1;i<n;i++){
            if(nums[i]==nums[i-1]+1){
                sum+=nums[i];
            }else{
                break;
            }
        }
        while(st.count(sum)){
            sum++;
        }
        return sum;
        
    }
};

File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/2996-smallest-missing-integer-greater-than-sequential-prefix-sum.py
class Solution(object):
    def missingInteger(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n=len(nums)
        st=set(nums)
        sq=nums[0]
        for i in range(1,n):
            if(nums[i]==nums[i-1]+1):
                sq+=nums[i]
            else:
                break
        
        while(sq in st):
            sq+=1
        return sq
        